### PR TITLE
Avoid using Pattern and Streams in UnionPath init

### PR DESCRIPTION
**Pull request by ichttt**
Speeds up creation of UnionPath significantly. Reduces time spend in testJarFileExists from 7585 to 2777, other tests like testCommonPathUtilities see improvements as well

Benchmark results on my Windows machine:
Before:
"Benchmark","Mode","Threads","Samples","Score","Score Error (99,9%)","Unit"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testByteChannel","avgt",1,2,"129528,239577",NaN,"ns/op"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testByteChannel:·jfr","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testByteChannel:·stack","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testCommonPathUtilities","avgt",1,2,"2310,161589",NaN,"ns/op"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testCommonPathUtilities:·jfr","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testCommonPathUtilities:·stack","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testDirectoryStream","avgt",1,2,"91979,226766",NaN,"ns/op"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testDirectoryStream:·jfr","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testDirectoryStream:·stack","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testDirectoryStreamDir","avgt",1,2,"206381,228068",NaN,"ns/op"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testDirectoryStreamDir:·jfr","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testDirectoryStreamDir:·stack","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testJarFileExists","avgt",1,2,"7585,715661",NaN,"ns/op"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testJarFileExists:·jfr","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testJarFileExists:·stack","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testJarFileNotExists","avgt",1,2,"7958,457813",NaN,"ns/op"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testJarFileNotExists:·jfr","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testJarFileNotExists:·stack","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testNativeFileExists","avgt",1,2,"120012,356044",NaN,"ns/op"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testNativeFileExists:·jfr","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testNativeFileExists:·stack","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testNativeFileNotExists","avgt",1,2,"81349,792250",NaN,"ns/op"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testNativeFileNotExists:·jfr","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testNativeFileNotExists:·stack","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testReadAttributes","avgt",1,2,"7883,473942",NaN,"ns/op"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testReadAttributes:·jfr","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testReadAttributes:·stack","avgt",1,1,NaN,NaN,"---"

After:
"Benchmark","Mode","Threads","Samples","Score","Score Error (99,9%)","Unit"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testByteChannel","avgt",1,2,"120842,178430",NaN,"ns/op"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testByteChannel:·jfr","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testByteChannel:·stack","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testCommonPathUtilities","avgt",1,2,"386,721520",NaN,"ns/op"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testCommonPathUtilities:·jfr","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testCommonPathUtilities:·stack","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testDirectoryStream","avgt",1,2,"84422,986678",NaN,"ns/op"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testDirectoryStream:·jfr","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testDirectoryStream:·stack","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testDirectoryStreamDir","avgt",1,2,"212206,019691",NaN,"ns/op"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testDirectoryStreamDir:·jfr","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testDirectoryStreamDir:·stack","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testJarFileExists","avgt",1,2,"2777,622614",NaN,"ns/op"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testJarFileExists:·jfr","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testJarFileExists:·stack","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testJarFileNotExists","avgt",1,2,"3192,212457",NaN,"ns/op"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testJarFileNotExists:·jfr","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testJarFileNotExists:·stack","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testNativeFileExists","avgt",1,2,"116069,613764",NaN,"ns/op"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testNativeFileExists:·jfr","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testNativeFileExists:·stack","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testNativeFileNotExists","avgt",1,2,"78360,375172",NaN,"ns/op"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testNativeFileNotExists:·jfr","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testNativeFileNotExists:·stack","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testReadAttributes","avgt",1,2,"2930,187392",NaN,"ns/op"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testReadAttributes:·jfr","avgt",1,1,NaN,NaN,"---"
"cpw.mods.niofs.union.benchmarks.UnionFileSystemBenchmark.testReadAttributes:·stack","avgt",1,1,NaN,NaN,"---"

### CODE REVIEW

**Comment by cpw** [comment]
Looks good but going to double check before merging..

File: src/main/java/cpw/mods/niofs/union/UnionPath.java [hunk 1]
```diff
@@ -67,13 +61,22 @@ private UnionPath(final UnionFileSystem fileSystem, boolean absolute, boolean is
...
 
     private String[] getPathParts(final String longstring) {
-        var clean = longstring.replace("\\", UnionFileSystem.SEP_STRING);
-        clean = SEPARATOR_BEGIN_END.matcher(clean).replaceAll("");
-        clean = SEPARATOR_DUPLICATES.matcher(clean).replaceAll("");
-        if (clean.isEmpty())
-            return new String[0];
-        else
-            return SEPARATOR_SPLIT.split(clean);
+        var clean = longstring.replace('\\', '/');
```
Comment (pupnewfster) [inline src/main/java/cpw/mods/niofs/union/UnionPath.java:64]: Why the change to make this not use the constant anymore?

Comment (ichttt) [reply src/main/java/cpw/mods/niofs/union/UnionPath.java:64]: Cause I use replace with char, not with String. It's ever so slightly faster

Comment (cpw) [reply src/main/java/cpw/mods/niofs/union/UnionPath.java:64]: yeah. this is fine and already merged as is.
